server/channels/api4/saml.go

```diff
@@ -46,7 +46,10 @@ func getSamlMetadata(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	w.Header().Set("Content-Type", "application/xml")
 	w.Header().Set("Content-Disposition", "attachment; filename=\"metadata.xml\"")
-	w.Write([]byte(metadata))
+	if _, err := w.Write([]byte(metadata)); err != nil {
+		c.Err = model.NewAppError("getSamlMetadata", "api.admin.saml.get_metadata.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
```

Comment:
If `w.Write` fails, there is likely an issue with the HTTP connection. It's not useful to then send an error to the client. Let's just log the error.